### PR TITLE
Fix pillar issue in public cloud

### PR DESCRIPTION
**Pull request by jgleissner**
I've added a call_runner function to salt interface and added a call to sync_pillar runner in cloud_cluster.rb. This is required to activate the external pillar, which is normally done at orchestration time, but for creating cluster nodes in the cloud, the pillar is needed before that. See also #399 for details.

I've also added a sleep(1) after the refresh_pillar, as we ran into the issue that the succeeding salt-cloud still works on the old pillar data, and this seems to work around it. I know that this is a bit ugly and perhaps can be fixed better in salt directly, but I haven't had any luck yet with narrowing the issue down.

### CODE REVIEW

**Comment by jordimassaguerpla** [comment]
Hi @jgleissner ! Could you check the failing tests in travis?

**Comment by jgleissner** [comment]
Hi, I'm looking into it. I believe it's because I didn't describe the call_runner function in spec/lib/velum/salt_spec.rb. I'm not really familiar with VCR cassettes and all that. It'll take me a bit to get up to speed on this. Sorry about that.

**Comment by jgleissner** [comment]
Hi @jordimassaguerpla , is there an easy way to run that travis test locally?

**Comment by jgleissner** [comment]
Don't worry about it, I think I got the test suite running ok.

**Comment by ereslibre** [comment]
However, travis is red...

File: app/models/cloud_cluster.rb [hunk 1]
```diff
@@ -66,7 +66,9 @@ def save!
     persist_to_pillar!(:cloud_worker_net, @network_id)
     persist_to_pillar!(:cloud_worker_subnet, @subnet_id)
     persist_to_pillar!(:cloud_worker_security_group, @security_group_id)
-    Velum::Salt.call(action: "saltutil.refresh_pillar")
+    Velum::Salt.call_runner(action: "saltutil.sync_pillar")
+    Velum::Salt.call(targets: "admin", action: "saltutil.refresh_pillar")
+    sleep(1)
```
Comment (MaximilianMeister) [inline app/models/cloud_cluster.rb]: why do we need the sleep(1)? maybe add a comment

Comment (jgleissner) [reply app/models/cloud_cluster.rb]: Makes sense, I'll add one.

**Comment by jgleissner** [comment]
Regarding the failing test, I believe this is due to the new runner call that does a http request to 127.0.0.1/run, which wasn't the case before, so the recorded cassette doesn't match anymore. I've been trying to figure out how this stuff works and made some progress, but haven't managed to record a new one yet. Sorry about that. If you can point me to any useful documentation etc. I'd appreciate it.

**Pull request merged**